OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableBoolean(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableDoubleReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyDouble = "varDouble";
+            var variableKeyInt = "varInt";
+            var variableKeyNonDouble = "varNonDouble";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.DOUBLE;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyDouble, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100.54");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyInt, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNonDouble, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("non_double_value");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual(100.54, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyDouble, TestUserId, null));
+            Assert.AreEqual(100, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyNonDouble, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableDouble(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableIntegerReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyInt = "varInt";
+            var variableKeyDouble = "varDouble";
+            var variableNonInt = "varNonInt";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.INTEGER;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyInt, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyDouble, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("100.45");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableNonInt, It.IsAny<string>(),
```

Comment:
the best way is first mock & then apply assert, then mock next & apply assert.
Please revise for me ;)

Reply:
Done